assignments/lambda-classes.js

```diff
@@ -1 +1,145 @@
 // CODE here for your Lambda Classes
+
+// Base Person Class
+
+class Person {
+    constructor(attributes){
+        this.name = attributes.name;
+        this.age = attributes.age;
+        this.location = attributes.location;
+    };
+    speak(){
+        return `Hello my name is ${this.name}, I am from ${this.location}.`;
+    };
+};
+
+// Instructor Class
+
+class Instructor extends Person {
+    constructor(instAttributes){
+        super(instAttributes);
+        this.specialty = instAttributes.specialty;
+        this.favLanguage = instAttributes.favLanguage;
+        this.catchPhrase = instAttributes.catchPhrase;
+    };
+    demo(subject){
+        return `Today we are learning about ${subject}.`
+    };
+    grade(student, subject){
+        return `${student.name} receives a perfect score on ${subject}.`
+    };
+    changeGrade(student){
+        let sign = Math.random() < 0.5 ? -1 : 1;
+        let change = sign * (Math.floor(Math.random()*10)); // Since Math.random generates a value between 0 and 1, we multiply by 10 and take floor to get an integer between 0 and 10. We then multiply by the sign (-1 or 1) which we calculated above. 
+        student.grade += change;
```

Comment:
You could also try this way    randomGrade(student) {
       const random = (x) => Math.ceil(Math.random() * 1.5 * x);
       student.grade = random(student.grade);
       return  `${this.name} modified ${student.name}\'s grade to be ${student.grade}...`;
   }

randomGrade(student) {
       const random = (x) => Math.ceil(Math.random() * 1.5 * x);
       student.grade = random(student.grade);
       localStorage.setItem('grade${student.name}')
       return  `${this.name} modified ${student.name}\'s grade to be ${student.grade}...`;
   }
try this out if you get a chance let jack = new Student({
   name: 'Jack',
   age: 18,
   location: 'Dallas, Texas',
   gender: 'Male',
   previousBackground: 'High School',
   className: 'WEB18',
   favSubjects: ['HTML',
       'CSS',
       'JavaScript'
   ],
   grade: localStorage.getitem('grade${this.name}') ?  localStorage.getitem('grade${this.name}') : 70
})